src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,41 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import java.util.Collections;
+import java.util.List;
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T>  {
+
+    @Override
+    public List<? extends T> getAll(Class<? extends T> type) {
+        if (type == Bulldozer.class) {
+            MachineProducer<Bulldozer> producer = new BulldozerProducer();
+            return (List<? extends T>) producer.get();
+        } else if (type == Excavator.class) {
+            MachineProducer<Excavator> producer = new ExcavatorProducer();
+            return (List<? extends T>) producer.get();
+        } else if (type == Truck.class) {
+            MachineProducer<Truck> producer = new TruckProducer();
```

Comment:
There is a missing import for `TruckProducer`. You need to import `core.mate.academy.service.TruckProducer` to avoid a compilation error.